_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,217 @@
+---
+title: "Stay on topic: Using Kafka to trigger your OpenFaaS functions"
+description: Martin outlines how you can use Kafka and subscribe OpenFaaS function to a topic using kafka-connector plugin.
+date: 2019-03-14
+image: /images/google-form-response/dots.jpg
+categories:
+  - kafka
+  - tutorial
+  - examples
+author_staff_member: martin
+dark_background: true
+---
+
+In this post I will show you how to subscribe your OpenFaaS functions to Kafka topics when running your functions via Kubernetes.
+
+OpenFaaS is production ready Serverless framework, and often finds itself as a part of bigger eco-system of software where the disparate parts may communicate with each-other in different ways. The OpenFaaS gateway exposes REST endpoints, but what if our services don't use HTTP?
+
+Kafka is stream-processing platform and at a high level the important concepts are Producers, Consumers and Brokers which communicate with each-other via massages attached to Topics. The way it works is that the Producer is sending messages on specific Topics to the Broker, then the Broker notifies all the Consumers which are interested in that Topic. The approach is fitting for distributed systems, because it decouples the communication between your services. 
```

Comment:
> then the Broker notifies all the Consumers which are interested in that Topic
Nit pick: technically, the broker does not notify the consumers like in a pub/sub system. Consumers (long) poll the topics.

Reply:
Thanks Michael 👍 Added that the connector actually polls the topics